contracts/BridgeExecutor.sol

```diff
@@ -0,0 +1,139 @@
+//SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "./lib/Pausable.sol";
+import "./SharedStructs.sol";
+import "./lib/BoolTokenTransfer.sol";
+import "./access/AdminRole.sol";
+import "./access/BridgeRole.sol";
+
+contract BridgeExecutor is Initializable, Pausable, BridgeRole {
+    using BoolTokenTransfer for IERC20;
+
+    /*========================= CONTRACT STATE =========================*/
+    uint128 public constant MIN_GAS_LIMIT_FOR_SC_CALL = 10_000_000;
+    uint128 public constant DEFAULT_GAS_LIMIT_FOR_REFUND_CALLBACK = 20_000_000;
+    uint256 private lowestTxId;
+    uint256 private currentTxId;
+
+    mapping(uint256 => MvxTransaction) private pendingTransactions;
+
+    /*========================= PUBLIC API =========================*/
+    function initialize() public initializer {
+        __BridgeRole_init();
+        __Pausable_init();
+    }
+
+    function __BridgeProxy__init_unchained() internal onlyInitializing {
+        lowestTxId = 0;
+        currentTxId = 0;
+    }
+
+    function deposit(MvxTransaction calldata txn) external payable whenNotPaused onlyBridge returns (bool) {
+        pendingTransactions[currentTxId] = txn;
+        currentTxId++;
+
+        IERC20 token = IERC20(txn.token);
+        bool approvalSuccess = token.approve(txn.recipient, txn.amount);
+
+        return approvalSuccess;
+    }
+
+    function execute(uint256 txId) external whenNotPaused {
+        require(txId < currentTxId, "BridgeExecutor: Invalid transaction ID");
+        MvxTransaction memory txn = pendingTransactions[txId];
+
+        if (txn.callData.length > 0) {
+            (bytes memory selector, uint256 gasLimit, bytes memory args) = abi.decode(
+                txn.callData,
+                (bytes, uint256, bytes)
+            );
+
+            if (selector.length == 0 || gasLimit == 0 || gasLimit < MIN_GAS_LIMIT_FOR_SC_CALL) {
+                _refundAndDeleteTxn(txId);
+                return;
+            }
+
+            bytes memory data;
+            if (args.length > 0) {
+                data = abi.encodePacked(selector, args);
+            } else {
+                data = selector;
+            }
+
+            _updateLowestTxId();
+
+            address _token = txn.token;
```

Comment:
I think these two assignments are redundant. We can use `txn.token` and `txn.amount` in the `_refundTransaction(_token, _amount);` call